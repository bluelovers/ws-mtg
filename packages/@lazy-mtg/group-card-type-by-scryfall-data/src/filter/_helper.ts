import { ITSAndTypeAndStringLiteral } from 'ts-type/lib/helper/string';
import { ITSArrayListMaybeReadonly, ITSValueOrArrayMaybeReadonly } from 'ts-type/lib/type/base';

export function dfGroupArrayFieldByPriority<T extends string>(priority: ITSArrayListMaybeReadonly<T>,
	defaultValue?: ITSAndTypeAndStringLiteral<T>,
)
{
	return (arrayValue: ITSValueOrArrayMaybeReadonly<ITSAndTypeAndStringLiteral<T>>) =>
	{
		let returnValue: T = defaultValue as any;

		arrayValue = [arrayValue].flat();

		priority.find(type =>
		{
			if (arrayValue.includes(type))
			{
				returnValue = type
				return true;
			}
		});

		return returnValue
	}
}

export function dfGroupFieldByPriority<T extends string>(priority: ITSArrayListMaybeReadonly<T>,
	defaultValue?: ITSAndTypeAndStringLiteral<T>,
)
{
	return (inputValue: ITSAndTypeAndStringLiteral<T>) =>
	{
		let returnValue: T = defaultValue as any;

		priority.find(type =>
		{
			if (inputValue === type)
			{
				returnValue = type
				return true;
			}
		});

		return returnValue
	}
}
