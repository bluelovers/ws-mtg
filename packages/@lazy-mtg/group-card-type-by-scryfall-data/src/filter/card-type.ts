
import { GROUP_CARD_TYPE_PRIORITY } from '../const';
import { dfGroupArrayFieldByPriority } from './_helper';

export const groupKeyByCardTypePriority = dfGroupArrayFieldByPriority(GROUP_CARD_TYPE_PRIORITY);
