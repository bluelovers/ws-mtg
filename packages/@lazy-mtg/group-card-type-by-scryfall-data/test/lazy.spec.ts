import { groupKeyByCardTypePriority } from '../src/filter/card-type';

test(`test`, () =>
{

	let actual = groupKeyByCardTypePriority([
		'Land',
		'Creature',
	]);

	expect(actual).toMatchSnapshot();

	actual = groupKeyByCardTypePriority('Land');

	expect(actual).toMatchSnapshot();

});


