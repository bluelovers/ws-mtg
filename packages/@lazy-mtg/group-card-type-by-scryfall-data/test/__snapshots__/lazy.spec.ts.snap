// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `"Creature"`;

exports[`test 2`] = `"Land"`;
